app/controllers/concerns/filter_parameters.rb

```diff
@@ -25,7 +19,33 @@ def deserialize_array_filter_param(param)
     params[param].split(",") if params.key?(param)
   end
 
+  def remove_previous_parameters(all_parameters)
+    all_parameters.except(*previous_parameters.keys)
+  end
+
+  def merge_previous_parameters(all_parameters)
+    previous_parameters.each do |key, value|
+      next if key =~ /prev_/ && value == "none"
+
+      all_parameters[key.split("_").last] = value
+    end
+
+    remove_previous_parameters(all_parameters)
+  end
+
+private
+
   def serialize_array_filter_param(value)
     value.join(",") if value.present?
   end
+
+  def parameters
+    return request.query_parameters if request.method == "GET"
+
+    request.request_parameters
+  end
+
+  def previous_parameters
+    parameters.select { |key, _value| key.to_s =~ /^prev_/ }
```

Comment:
Wonder if there's a nice way to refactor this regex that's in a lot of places too, maybe moving it to a contatnt like `PREVIOUS_PARAMETER_NAME_MATCHER` or something

Reply:
Also maybe worth using the method-for-the-job:

```suggestion
    parameters.select { |key, _value| key.to_s.start_with? "^prev_" }
```